.github/workflows/ccpp.yml

```diff
@@ -110,7 +116,148 @@ jobs:
           set -x
           exec 2>&1
           abs_top_builddir="$PWD"
+          pkg-config --version
+          pkg-config --help
+          pkg-config --about ||:
+          pkg-config --path libexif ||:
+          pkg-config --path libgphoto2 ||:
+          export PKG_CONFIG_PATH="${abs_top_builddir}/__prefix/lib/pkgconfig:$PKG_CONFIG_PATH"
+          pkg-config --path libexif ||:
+          pkg-config --path libgphoto2 ||:
+          export PKG_CONFIG_PATH="${abs_top_builddir}/__prefix/lib/pkgconfig:"
+          pkg-config --path libexif ||:
+          pkg-config --path libgphoto2 ||:
+          export PKG_CONFIG_PATH="${abs_top_builddir}/__prefix/lib/pkgconfig"
+          pkg-config --path libexif ||:
+          pkg-config --path libgphoto2 ||:
+          export LD_LIBRARY_PATH="${abs_top_builddir}/__prefix/lib"
+          cd examples/ambs-lgp2-frontend
+          autoreconf -vis
+          ./configure --prefix="$PWD/__pref"
+          make
+          ${LDD-false} ambs-lgp2-frontend
+          ./ambs-lgp2-frontend
+          make install
+          ${LDD-false} __pref/bin/ambs-lgp2-frontend
+          __pref/bin/ambs-lgp2-frontend
+
+  msys2-build:
+    needs: ix-build
+
+    runs-on: ${{ matrix.os }}
+
+    name: '${{ matrix.os }} ${{ matrix.sys }}'
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest]
+        # FIXME: mingw32 cannot find WSAStartup() in -lws2_32
+        sys: [mingw64, ucrt64, clang64]
+
+    defaults:
+      run:
+        shell: 'msys2 {0}'
+
+    steps:
+      - name: 'git config core.autocrlf (Windows)'
+        run: git config --global core.autocrlf input
+        shell: bash
+
+      - uses: actions/checkout@v3
+
+      - name: 'Set MINGW_ARCH (Windows)'
+        shell: bash
+        run: echo MINGW_ARCH="${{ matrix.sys }}" >> $GITHUB_ENV
+
+      # See https://github.com/msys2/setup-msys2
+      - name: 'Set up MSYS2 and install build requirements (Windows)'
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{ matrix.sys }}
+          update: true
+          install: >-
+            git
+            autoconf
+            automake
+            gettext-devel
+            libtool
+            make
+            pkgconf
+            msys2-w32api-headers
+            msys2-w32api-runtime
```

Comment:
If building w/ autotools, better install just env specific `autotools:p` through pacman below, it'll pull in whatever needed.

You don't need msys2-w32api-* packages here, as nothing is being build for the msys2 env...

Here's the list of deps for the [MSYS2 repo package](https://packages.msys2.org/package/mingw-w64-ucrt-x86_64-libgphoto2?repo=ucrt64).

So in this section, you basically just need `git`, all the other ones belong to the env specific list in pacboy section below.

Reply:
We link against `ws2_32.dll` and `msys2-w32api-runtime` contains a `ws2_32.dll`, so I thought `msys2-w32api-runtime` was the package to install. Could you tell me where my train of thought derails here?

`msys2-w32api-headers` is a leftover from trying to fix the mingw32 build which does not find the `WSAstartup()` function in `ws2_32.dll`. However, it did not help, so I can indeed remove it.

Reply:
`ws2_32.dll` is a Windows system library provided by MS, no need to install it.

Basically, msys2 is a POSIX emulation environment (a fork of Cygwin), while mingw64 etc. are native Windows environments one wants to build for. Apart from a couple of choice msys2 tools, one really doesn't need any more development headers and libraries there, everything must come from the native Windows environments.